hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/heartbeat/HoodieHeartbeatClient.java

```diff
@@ -266,6 +266,11 @@ private void updateHeartbeat(String instantTime) throws HoodieHeartbeatException
       heartbeat.setLastHeartbeatTime(newHeartbeatTime);
       heartbeat.setNumHeartbeats(heartbeat.getNumHeartbeats() + 1);
     } catch (IOException io) {
+      Boolean isHeartbeatStopped = instantToHeartbeatMap.get(instantTime).isHeartbeatStopped;
+      if (isHeartbeatStopped) {
+        LOG.warn(String.format("update heart beat failed, because the instant time %s was stopped ? : %s", instantTime, isHeartbeatStopped));
+        return;
+      }
```

Comment:
is it possible to write a test for this?

Reply:
@nsivabalan 
It is difficult that the asynchronous heartbeat Timer task thread sends a request to update the heartbeat file to hdfs when the write client stops the heartbeat.
I can only write a similar unit test as follows. Do you have any suggestions?
```
@Test
  public void testUpdateStoppedHeartbeat() {

    HoodieHeartbeatClient hoodieHeartbeatClient =

        new HoodieHeartbeatClient(metaClient.getFs(), metaClient.getBasePath(), heartBeatInterval, numTolerableMisses);

    hoodieHeartbeatClient.start(instantTime1);

    hoodieHeartbeatClient.stopHeartbeatTimers();

    assertDoesNotThrow(() -> hoodieHeartbeatClient.updateHeartbeat(instantTime1));
  }
```